pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +49,51 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
+// cgroup.
+func cgrpSanityCheck() {
+
+	procCount := 0
+	err := filepath.Walk(cgroupRoot, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: error traversing cgroup directory")
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if info.Name() != "cgroup.procs" {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: failed to open file")
```

Comment:
Any time there is an informational log, I'd like to see that it is actionable information that the user can use. Otherwise we end up just littering the logs with random stuff that the user both doesn't need to know and can't do anything about. Is this level of detail really necessary to log at Info level or should we drop this to Debug?

I'm thinking, if the filepath walk hits a file that's not openable for some reason then the user probably can't do anything about this. Cgroup filesystem is a pretty specialized thing that will either work or not. It's not like we expect the user to go and change permissions or create a file in this path or something.

Same feedback for the other Info log from this `filepath.Walk()`.

Reply:
That makes sense. I went back and forth into adding this. My rationale for keeping it was as follows: if I'm debugging an issue it would be helpful to know if the warning was not hit because the check was  successful  or because some other (bogus) error happened and we had to bail out. I agree that it would not be useful from a user perspective (this is why I opted for an info level), but if a developer is looking into a log from a sysdump, I think it will be useful.

That being said, I'm happy to remove it if the above reason is not enough to keep it.

Reply:
Update: removed the warnings. I'd be interested to know what you think regarding the above comment though.

Reply:
It's always a balance :thinking: 

When I read these the first time, I saw that this would print an info level message then return `nil` so then it won't even look like there was any warning or error condition that should be reported by the user. It was just a random info message without any sort of prompt to the user that they should report something. The rest of the logic of Cilium would continue as though there wasn't any problem after this `filepath.Walk()` execution. Additionally given the message, it looked like developer log messages which are only useful if we effectively communicate to users that we want to see these. If we are really worried about this condition, then maybe it shouldn't be at a lower severity level like I suggested (ie Debug), but instead higher (Warning) and tell the user what we'd like, for example:

```
log.WithFields(logrus.Fields{
    logfields.Path: path,
    "area": "cgroup-sanity",
    logfields.URL: "https://github.com/cilium/cilium/issues",
}).WithError(err).Warning("Cgroup sanity check failed to open path. Please report this issue on Github with information about your environment")
```

On a separate note, on my wishlist for a while is to have dynamic per-subsystem log level adjustment. Then we could set these to debug level and only if we suspect there's a problem in cgroups area we can set the cgroups debug to a higher verbosity level and then we can (1) get the info we need to debug and (2) won't get drowned out by all of the verbose logs from other subsystems... but that'll take a bit more invasive trickery in the logging package to get there I think.